Makefile

```diff
@@ -10,7 +10,9 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-OS := $(shell uname -s)
+ifndef OS
```

Comment:
Why is this needed? If this variable is already defined it could be already used for something else and might not be what we expect. (E.g. if I maliciously do `export OS="Banana"` the script wouldn't work). We could use a more specific variable name if you think there could be a conflict?

Reply:
because environment variable `OS` is predefined on Windows. `OS` will be overwritten without this line by `uname` command